containers/daap-presigned-url/src/var/task/presigned_url.py

```diff
@@ -20,30 +20,46 @@
 def handler(event, context):
     data_product_name = event["pathParameters"].get("data-product-name")
     table_name = event["pathParameters"].get("table-name")
-    md5 = str(event["body"].get("contentMD5"))
+    body = json.loads(event.get("body"))
+    md5 = str(body.get("contentMD5"))
+    filename = body.get("filename")
     amz_date = datetime.utcnow()
     formatted_date = amz_date.strftime("%Y%m%dT%H%M%SZ")
     uuid_value = uuid.uuid4()
 
-    if not isinstance(data_product_name, str) or not isinstance(table_name, str):
+    if (
+        not isinstance(data_product_name, str)
+        or not isinstance(table_name, str)
+        or not isinstance(filename, str)
+    ):
         return {
             "statusCode": 400,
             "headers": {"Content-Type": "application/json"},
             "body": json.dumps(
                 {
                     "error": {
-                        "message": "data product name or table name value are not"
+                        "message": "data product name, table name or filename are not"
                         + " convertible to string type."
                     }
                 }
             ),
         }
 
+    file_extension = os.path.splitext(filename)[1]
```

Comment:
```suggestion
    from pathlib import PurePath  # top of file import if using this suggestion
    file_extension = os.path.splitext(filename)[1]
    file_extension = PurePath.suffix(filename)
```
I highly recommend the pathlib module for interacting with paths. As we're not actually representing an OS path, use PurePath in place of Path

Reply:
There's also a way to augment/extend this later for more complex suffixes e.g. `.SNAPPY.parquet`, using Path.suffixes, which collects all suffixes on a file.

Reply:
Now using PurePath as suggested 👍🏻